artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/connect/AMQPBrokerConnection.java

```diff
@@ -432,11 +432,29 @@ private void doConnect() {
                   final boolean coreTunnelingEnabled = isCoreMessageTunnelingEnabled(replica);
                   final Symbol[] desiredCapabilities;
 
-                  if (coreTunnelingEnabled) {
-                     desiredCapabilities = new Symbol[] {AMQPMirrorControllerSource.MIRROR_CAPABILITY,
-                                                         AmqpSupport.CORE_MESSAGE_TUNNELING_SUPPORT};
-                  } else {
-                     desiredCapabilities = new Symbol[] {AMQPMirrorControllerSource.MIRROR_CAPABILITY};
+                  if (!replica.isForward()){
```

Comment:
Once you get to this level of complexity to manage the elements in the desired capabilities I'd suggest just switch to creating an ArrayList and adding the elements individually based on the configuration and then just doing a toArray() on that to get the final capabilities array.